engine/src/main/java/org/hibernate/search/query/dsl/sort/MultiValuedMode.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ *
+ */
+package org.hibernate.search.query.dsl.sort;
+
+/**
+ * Mode when comparing multiple values.
+ *
+ * @author Emmanuel Bernard emmanuel@hibernate.org
+ */
+public enum MultiValuedMode {
+	/**
+	 * Pick the lowest value.
+	 */
+	MIN,
+
+	/**
+	 * Pick the highest value.
+	 */
+	MAX,
+
+	/**
+	 * Use the sum of all values as sort value.
+	 * Only applicable for number based array fields.
```

Comment:
"Only applicable for number based array fields"

I guess the focus should be on *number* but not strictly only an *array* ? Could work also on numeric values which have been indexed as strings via some string bridge?

Comment applies on all three of SUM, AVG, MEDIAN.

BTW as commented on JIRA, in this case it seems like you're also adding a new feature (on top of a new API).. feel free to split this up into a follow up task? I think currently our docs are pretty clear that you can't sort on multi-valued fields.